src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1040,27 +1043,46 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
             case (AST::operatorType::FloorDiv) : {op = ASR::binopType::Div; break;}
             case (AST::operatorType::Pow) : { op = ASR::binopType::Pow; break; }
             case (AST::operatorType::Mod) : { is_mod = true; break; }
+            case (AST::operatorType::BitOr) : { is_bitwise_or = true; break; }
+            case (AST::operatorType::BitAnd) : { is_bitwise_and = true; break; }
+            case (AST::operatorType::BitXor) : { is_bitwise_xor = true; break; }
             default : {
                 throw SemanticError("Binary operator type not supported",
                     x.base.base.loc);
             }
         }
+        left = implicitcast_helper(ASRUtils::expr_type(right), left);
+        right = implicitcast_helper(ASRUtils::expr_type(left), right);
+        Vec<ASR::call_arg_t> args;
+        args.reserve(al, 2);
+        ASR::call_arg_t arg1, arg2;
+        arg1.loc = left->base.loc;
+        arg1.m_value = left;
+        args.push_back(al, arg1);
+        arg2.loc = right->base.loc;
+        arg2.m_value = right;
+        args.push_back(al, arg2);
+
         if (is_mod) {
-            left = implicitcast_helper(ASRUtils::expr_type(right), left);
-            right = implicitcast_helper(ASRUtils::expr_type(left), right);
             ASR::symbol_t *fn_mod = resolve_intrinsic_function(x.base.base.loc, "_mod");
-            Vec<ASR::call_arg_t> args;
-            args.reserve(al, 2);
-            ASR::call_arg_t arg1, arg2;
-            arg1.loc = left->base.loc;
-            arg1.m_value = left;
-            args.push_back(al, arg1);
-            arg2.loc = right->base.loc;
-            arg2.m_value = right;
-            args.push_back(al, arg2);
             tmp = make_call_helper(al, fn_mod, current_scope, args, "_mod", x.base.base.loc);
             return;
         }
+        if (is_bitwise_or) {
+            ASR::symbol_t *fn_mod = resolve_intrinsic_function(x.base.base.loc, "_bitwise_or");
+            tmp = make_call_helper(al, fn_mod, current_scope, args, "_bitwise_or", x.base.base.loc);
+            return;
+        }
+        if (is_bitwise_and) {
+            ASR::symbol_t *fn_mod = resolve_intrinsic_function(x.base.base.loc, "_bitwise_and");
+            tmp = make_call_helper(al, fn_mod, current_scope, args, "_bitwise_and", x.base.base.loc);
+            return;
+        }
+        if (is_bitwise_xor) {
+            ASR::symbol_t *fn_mod = resolve_intrinsic_function(x.base.base.loc, "_bitwise_xor");
+            tmp = make_call_helper(al, fn_mod, current_scope, args, "_bitwise_xor", x.base.base.loc);
+            return;
+        }
```

Comment:
```suggestion
            ASR::symbol_t *fn_mod = resolve_intrinsic_function(x.base.base.loc, bitwise_op_name);
            tmp = make_call_helper(al, fn_mod, current_scope, args, bitwise_op_name, x.base.base.loc);
            return;
```

Reply:
Done, thanks!

Reply:
You will have to add a check to not return when `bitwise_op_name` is different from the three cases considered.